### DISCUSSION TITLE
Download from bucket using `google-storage1`

### DISCUSSION THREAD

**Discussion by 35359595** [Byron/google-apis-rs#494] [Q&A]
Need some direction on how to join object listing and downloading.
Here's what I've tried so far (gives 404 on second call - `.list()` works ok):

```rust
  for obj in hub
            .objects()
            .list(&bucket)
            .doit()
            .await?
            .1
            .items
            .unwrap()
        {
            let name = obj.name.unwrap();
            let block_json = hub
                .objects()
                .get(&bucket, &name)
                .param("alt", "media")
                .doit()
                .await?
                .0
                .into_body()
                .data()
                .await
                .unwrap()?
                .to_vec();
        }
```

**Comment by Byron** [comment]
It's probably best to [consult the original API documentation](https://cloud.google.com/storage/docs/json_api) for examples and see how these map back to the generated API.
More specific help is impossible to provide, as this project is about the generator for APIs, not the APIs themselves. Thanks for your understanding.
